cluster/cluster.go

```diff
@@ -0,0 +1,127 @@
+package cluster
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/groupcache/groupcache-go/v3"
+	"github.com/groupcache/groupcache-go/v3/transport"
+	"github.com/groupcache/groupcache-go/v3/transport/peer"
+)
+
+var _daemons []*Daemon
+var _peers []peer.Info
+
+// ListPeers returns a list of all peers in the cluster
+func ListPeers() []peer.Info {
+	return _peers
+}
+
+// ListDaemons returns a list of all daemons in the cluster
+func ListDaemons() []*Daemon {
+	return _daemons
+}
+
+// DaemonAt returns a specific daemon
+func DaemonAt(idx int) *Daemon {
+	return _daemons[idx]
+}
+
+// PeerAt returns a specific peer
+func PeerAt(idx int) peer.Info {
+	return _peers[idx]
+}
+
+// FindOwningDaemon finds the daemon which owns the key provided
+func FindOwningDaemon(key string) *Daemon {
+	c, isRemote := _daemons[0].GroupCache.PickPeer(key)
+	if !isRemote {
+		return _daemons[0]
+	}
+
+	for i, d := range _daemons {
+		if d.ListenAddress() == c.PeerInfo().Address {
+			return _daemons[i]
+		}
+	}
+	panic(fmt.Sprintf("failed to find daemon which owns '%s'", key))
```

Comment:
I will suggest we figure out a way to use a logger interface with a panic function.

Reply:
I don't understand, a daemon cluster is mostly to be used in testing unless you have some other use case in mind. Panic should be fine here. Logging would be important if this is expected to be run in production.

Reply:
ok the purpose of the cluster is to use it in test. So the name cluster can be a confusing. Maybe let'us document as such. Because at first glance I can just spawn a group cache cluster in production. Also I thought we were adding this struct with some options to help start a cluster of nodes.

Reply:
@thrawn01 I am off the opinion anything related to test should go into some `testkit` package.

Reply:
It is intended for Developers who use groupcache in their projects to use `cluster` in their tests. It is a convenient way to start a cluster suitable for testing. As such `cluster` is a part of our public interface and should be treated as such.

I will revisit the documentation, and ensure that it is well understood, how this is to be used.

Reply:
> ok the purpose of the cluster is to use it in test. So the name cluster can be a confusing. Maybe let'us document as such. Because at first glance I can just spawn a group cache cluster in production. Also I thought we were adding this struct with some options to help start a cluster of nodes. 

Running a local cluster in production doesn't make much sense, a local cluster is only useful for testing.